lms/views/admin/application_instance.py

```diff
@@ -294,9 +294,35 @@ def show_instance(self):
     def update_instance(self):
         ai = self._get_ai_or_404(**self.request.matchdict)
 
-        for ai_field in ["lms_url", "deployment_id"]:
-            if value := self.request.params.get(ai_field, "").strip():
-                setattr(ai, ai_field, value)
+        if lms_url := self.request.params.get("lms_url", "").strip():
+            ai.lms_url = lms_url
+
+        if deployment_id := self.request.params.get("deployment_id", "").strip():
```

Comment:
This is a mouthful made a little bit longer by using the exception on, the service (rather than `one_or_none`).